src/App.js

```diff
@@ -2,46 +2,75 @@ import './styles/reset.css'
 import './styles/index.css'
 
 import initialStoreItems from './store-items'
-
-/*
-Here's what a store item should look like
-{
-  id: '001-beetroot',
-  name: 'beetroot',
-  price: 0.35
-}
-
-What should a cart item look like? 🤔
-*/
+import Store from './Store'
+import { useState } from 'react'
+import Cart from './Cart'
+import Total from './Total'
 
 console.log(initialStoreItems)
 
 export default function App() {
   // Setup state here...
+  const [cart, setCart] = useState([])
+  //const [quantity, setQuantity]= useState(1)
 
+  //  Adding item into Cart
+  function addToCart(item) {
+    // check if the item is already in the cart
+    // if it is increase the quantity by 1 and return
+    const itemFound = cart.find(itemFromCart => itemFromCart.id === item.id)
+    if (itemFound) {
+      const others = cart.filter(itemFromCart => itemFromCart.id !== item.id)
+      itemFound.quantity += 1
+      setCart([...others, itemFound])
```

Comment:
Will this move the item that you increased the quantity of, to the last place?

Reply:
its filtering the rest of the item in the cart and increasing the item added by 1 each time add to cart function called.